tools/tags/tags.go

```diff
@@ -44,6 +44,40 @@ func (and AndSet) Lines() (ls []string) {
 	return
 }
 
+// AggregateLines aggregates all tags from a set of files into one line, and remove duplicate tags.
+func (and AndSet) AggregateLines() (ls []string) {
+	var tags []string
+
+	// Aggregates all tags from a set of files, and remove duplicate tags.
+	for _, or := range and {
+		for i := 0; i < len(or); i++ {
+			dup := 0
+
+			for _, tag := range tags {
```

Comment:
This is clearly semantically different from the original... It is aggregated a series of "and" relations into a bunch of ORs. E.g. it turns

// +build x y
// +build a b

Which means (x || y) && (a || b)

Into:

x || y || a || b

I will revisit the tagging issue and see if there's a better fix.

Reply:
Note that I've pushed 1f4087e7cd6c3cc696e6b26446abd6c5214cfd67 to fix the specific go_marshal issue for ARM (it now works). I think beyond this the only _real_ way to fix these issues is to make go_marshal and go_stateify tag-aware and remove the tags package altogether. See also #2298.